app/login/route.js

```diff
@@ -0,0 +1,25 @@
+import Ember from 'ember';
+
+export default Ember.Route.extend({
+
+  actions: {
+
+    login() {
+
+      var controller = this.get('controller');
+      var email = controller.get('userEmail');
+      var password = controller.get('userPassword');
+
+      this.get('session').open('firebase', {
+        provider: 'password',
+        email: email,
+        password: password
+
+      }).then(function() {
+
+        this.transitionTo('protected');
+
+      }.bind(this));
```

Comment:
In ES6, there are [Arrow Functions](https://babeljs.io/learn-es2015/#ecmascript-2015-features-arrows-and-lexical-this). they preserve the current context of the current block of code in the newly created function.

```js
}).then(() => {
    this.transitionTo('protected');
}); 
```

Although `bind()`  can be used, I would stray away from using it.